src/content/tutorials/plugin/index.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Building Broccoli Plugins
+---
+
+The first question you should ask yourself is "what is a Broccoli plugin?".
+
+Plugins are what a build pipeline developer will interact with most. Plugins are what do the actual work of
+transforming files at each step of build process. The API of a plugin requires just 2 steps, creating a class that
+extends the [broccoli-plugin](https://github.com/broccolijs/broccoli-plugin) base class, and implementing a 
+`build()` method, that performs some work and/or returns a promise.
+
+${toc}
+
+Let's have a look at the basic building blocks of a Broccoli plugin.
+
+# Broccoli-Plugin base class
+
+```js
+const Plugin = require('broccoli-plugin');
+
+class MyPlugin extends Plugin
+{
+    constructor (inputNode, options) {
+        super([inputNodes], options);
+    }
+    
+    build() {
+        // A plugin can receive single or multiple inputs nodes/directories
+        // The files are available in this.inputPaths[0]/this.inputPaths[1]...
+        // You can do anything with these files that you can do in node
+        // The output of your plugin must write to this.outputPath
+    }
+}
+```
+
+## Constructor
+
+`inputNode`: This can be one of two things, a string or another Broccoli plugin. If a string is passed, Broccoli
+expects this to be a source directory and automatically converts this into a
+[broccoli-source](https://github.com/broccolijs/broccoli-source).
+
+When constructing your build pipeline you will be passing plugins as inputs to one another, these are commonly
+referred to as `inputNodes`. The `broccoli-plugin` base class expects an array of `inputNodes`, however the 
+majority of plugins will only require a single input, so feel free to pass an array up if only a single input is
+required.
+
+`options`: (all optional)
+
+```js
+{
+    name: '',
+    annotation: '',
+    persistentOutput: false,
+    needsCache: false,
+}
+```
+
+`options.name`: Custom name used when debugging/printing stack traces. Broccoli will use the name of your
+plugin if this field is not supplied.
+
+`options.annotation`: In addition to the plugin name, Broccoli uses annotation to provide a descriptive label used
+during debugging/printing stack traces. This is not often set by the plugin author, but by the consumer of the 
+plugin to tell multiple instances of the same plugin apart.
+
+`options.persistentOutput`: If true, the output directory is not automatically emptied between builds. By default,
+when the Broccoli process is stopped, all plugins temporary output directories are deleted. If `persistentOutput`
+is true, this will not happen and the contents will persist between processes. This allows a plugin which performs
```

Comment:
broccoli-plugin does not support persistence between process restarts, only persistent-filter and similar currently do. Future work, likely after we improve the `this.inputs` / `this.outputs` API, could expose this as a core competency

Reply:
cool, I misunderstood what this was for. Would you mind clarifying what i's for and why you'd use it?